cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/ServiceSpecificationCodec.java

```diff
@@ -72,8 +72,9 @@ public ServiceSpecification deserialize(JsonElement json, Type typeOfT,
                                                                     HttpServiceHandlerSpecification.class);
     Resources resources = context.deserialize(jsonObj.get("resources"), Resources.class);
     int instances = jsonObj.get("instances").getAsInt();
+    Map<String, String> properties = deserializeMap(jsonObj.get("properties"), context, String.class);
```

Comment:
does this return an empty map if properties is null? Need to protect against specs that we written by an older version of CDAP.

Reply:
Yes, it will return empty map